### PR TITLE
ATO-399/update logging config

### PR DESCRIPTION
**Pull request by CarlyG55**
## What?

Implements log4j and tidies up exception handling
## Why?

Log4j allows us to make our error messages cleaner. I've also updated our exception handling, because it wasn't catching runtime exceptions in the stub.

## Related PRs

Please include links to PRs in other repositories relevant to this PR.
Delete this section if not needed.

### CODE REVIEW

File: src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="INFO">
+    <Appenders>
+        <Console name="Console" target="SYSTEM_OUT">
+            <PatternLayout alwaysWriteExceptions="false" pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%replace{%ex}{[\r\n]{1,2}}{|}%n"/>
```
Comment (CarlyG55) [inline src/main/resources/log4j2.xml:5]: I appreciate that `alwaysWriteExceptions="false"` looks a bit strange, but the info is [here](https://stackoverflow.com/a/47760847)

File: src/main/java/uk/gov/di/handlers/ExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.apache.logging.log4j.Logger;
+import spark.Request;
+import spark.Response;
+import uk.gov.di.utils.ResponseHeaderHelper;
+import uk.gov.di.utils.ViewHelper;
+
+import java.util.HashMap;
+
+public class ExceptionHandler implements spark.ExceptionHandler<Exception> {
+    private static final Logger LOG = LogManager.getLogger(InternalServerErrorHandler.class);
```
Comment (hannah-leong) [inline src/main/java/uk/gov/di/handlers/ExceptionHandler.java:13]: Should this be:
```suggestion
    private static final Logger LOG = LogManager.getLogger(ExceptionHandler.class);
```

Comment (CarlyG55) [reply src/main/java/uk/gov/di/handlers/ExceptionHandler.java:13]: Yes, thank you!

**Review by hannah-leong — DISMISSED** [review]

File: src/main/java/uk/gov/di/handlers/AuthCallbackHandler.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 public class AuthCallbackHandler implements Route {
 
-    private static final Logger LOG = LoggerFactory.getLogger(AuthCallbackHandler.class);
+    private static final Logger LOG = LogManager.getLogger(AuthCallbackHandler.class);
```
Comment (ethanmills) [inline src/main/java/uk/gov/di/handlers/AuthCallbackHandler.java:19]: Can we keep the slf4j binding and wire it up to use log4j under the hood? https://www.slf4j.org/manual.html#swapping

I'm worried we'll lose framework logs like this

Comment (CarlyG55) [reply src/main/java/uk/gov/di/handlers/AuthCallbackHandler.java:19]: Done - yes we were missing some. It's a slightly different binding to the one you shared as I think that's just for log4j 1.x. I've added the changes as a new commit so hopefully it's easier to rereview, but happy to fix up the commit history for when it gets merged if you like

Comment (ethanmills) [reply src/main/java/uk/gov/di/handlers/AuthCallbackHandler.java]: LGTM, please squash away

**Review by ethanmills — APPROVED** [review]

**Pull request merged**